service/worker/scanner/executions/util.go

```diff
@@ -0,0 +1,89 @@
+// The MIT License (MIT)
+//
+// Copyright (c) 2017-2020 Uber Technologies Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package executions
+
+import (
+	"time"
+
+	"go.uber.org/cadence"
+	"go.uber.org/cadence/workflow"
+)
+
+func flattenShards(shards Shards) []int {
+	if len(shards.List) != 0 {
+		return shards.List
+	}
+	var result []int
+	for i := shards.Range.Min; i < shards.Range.Max; i++ {
+		result = append(result, i)
+	}
+	return result
+}
+
+func resolveFixerConfig(overwrites FixerWorkflowConfigOverwrites) ResolvedFixerWorkflowConfig {
+	resolvedConfig := ResolvedFixerWorkflowConfig{
+		Concurrency:             25,
+		BlobstoreFlushThreshold: 1000,
+		InvariantCollections: InvariantCollections{
+			InvariantCollectionMutableState: true,
+			InvariantCollectionHistory:      true,
+		},
+	}
+	if overwrites.Concurrency != nil {
+		resolvedConfig.Concurrency = *overwrites.Concurrency
+	}
+	if overwrites.BlobstoreFlushThreshold != nil {
+		resolvedConfig.BlobstoreFlushThreshold = *overwrites.BlobstoreFlushThreshold
+	}
+	if overwrites.InvariantCollections != nil {
+		resolvedConfig.InvariantCollections = *overwrites.InvariantCollections
+	}
+	return resolvedConfig
+}
+
+func getShortActivityContext(ctx workflow.Context) workflow.Context {
+	activityOptions := workflow.ActivityOptions{
+		ScheduleToStartTimeout: time.Minute,
+		StartToCloseTimeout:    time.Minute,
+		RetryPolicy: &cadence.RetryPolicy{
+			InitialInterval:    time.Second,
+			BackoffCoefficient: 1.7,
+			ExpirationInterval: 5 * time.Minute,
+		},
+	}
+	return workflow.WithActivityOptions(ctx, activityOptions)
+}
+
+func getLongActivityContext(ctx workflow.Context) workflow.Context {
+	activityOptions := workflow.ActivityOptions{
+		ScheduleToStartTimeout: time.Minute,
+		StartToCloseTimeout:    time.Minute,
```

Comment:
one minute for long running activity?

Reply:
Retry policy overwrites this value.

Reply:
Synced offline, this will be fixed in next pr.